### PR TITLE
[New Feature] Add new operation Gather

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

add new opteration gather

## How was this patch tested?
unit tests
manual tests on a textcnn model.

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Gather.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+class Gather[T: ClassTag, D: ClassTag]()(implicit ev: TensorNumeric[T], ev2: TensorNumeric[D])
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Gather.scala]: Please add java doc to the operation.

**Comment by yiheng** [comment]
Others looks good to me.

**Comment by qiuxin2012** [comment]
@yiheng Add float indices support, because we can't generate an integer indices tensor with optimizer's python API.

**Comment by yiheng** [comment]
LGTM

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**